### PR TITLE
Bug 1841214: Add bind address configuration setting

### PR DESCRIPTION
**Pull request by cybertron**
Previously coredns-mdns sent multicast traffic to all available
interfaces on the system. This is problematic in some environments
where it is undesirable to send such traffic to some interfaces.
For example, tunnel interfaces where a specific encapsulation is
required to avoid errors.

This change adds a new configuration option to the plugin. When
specified, the plugin will look up the interface associated with
the address and only send multicast traffic on that interface. If
the option is not specified, the behavior will be as before.

### CODE REVIEW

File: mdns.go [hunk 1]
```diff
@@ -165,8 +168,16 @@ func (m *MDNS) BrowseMDNS() {
 		}
 	}(srvEntriesCh)
 
-	queryService("_workstation._tcp", entriesCh)
-	queryService("_etcd-server-ssl._tcp", srvEntriesCh)
+	var iface net.Interface
+	if m.bindAddress != "" {
+		foundIface, err := publisher.FindIface(net.ParseIP(m.bindAddress))
+		if err != nil {
+			log.Errorf("Failed to find interface for '%s'\n", m.bindAddress)
```
Comment (frobware) [inline mdns.go:175]: Should this now return an error? I see we log but should we only assign `iface = foundIface` when this is nil?

File: README.md [hunk 2]
```diff
@@ -69,3 +69,19 @@ example.com {
...
+without setting a filter, set `filter text` to "".
+
+~~~ corefile
+example.com {
+    mdns example.com 3 "" 192.168.1.1
+}
+~~~
+
+This configuration will only send multicast packets to the interface assigned
+the `192.168.1.1` address. The interface lookup is dynamic each time an mDNS
```
Comment (frobware) [inline README.md:85]: How expensive is the lookup before each query is sent?

Comment (cybertron) [reply README.md:85]: Not very. It takes 1 to 2 ms on my system, which is insignificant compared to the time it takes to actually do the query.

Also worth noting that this runs in a background thread, so it doesn't impact response time of the DNS server at all.

File: mdns.go (hunk 1 above)
Comment (cybertron) [reply mdns.go:175]: Yep, you're right. I'll fix that.

**Comment by cybertron** [comment]
/cherry-pick release-4.5

We have to manually create branches in this repo. I've done that up to 4.5 now, so let's see if this works.

**Pull request merged**